### PR TITLE
chore: put regular and redhat-certified bundles into separate dirs

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

This PR tries to tackle the problem of differences between generated bundles due to different configs (e.g. redhat certified bundle needing related images patch).

This also:

- moves the `bundle.Dockerfile` that was present at the root of the repository into `bundle/regular/` dir
- adds 1 more Dockerfile, so that now we have 2:
  - `bundle/regular/bundle.Dockerfile` for "regular" builds
  - `bundle/redhat-certified/bundle.Dockerfile` for redhat certified builds, that have the `RELATED_IMAGE_*` env variables injected

**Special notes for your reviewer**:

~~This is sort of incomplete because `bundle.Dockerfile` - seems like not configurable in `operator-sdk generate bundle ...` - will get regenerated and changed depending on which bundle is being generated (either `bundle` or `bundle-redhat-certified`)~~

~~This needs to be addressed somehow (e.g. by adding manual renaming commands in `Makefile`?)~~

### CODE REVIEW

File: config/redhat-certified/related_images.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: kong-gateway-operator.v0.0.0
+  namespace: placeholder
+spec:
+  relatedImages:
+  - image: registry.connect.redhat.com/kong/kong:2.8.1
+    name: kong
+  - image: registry.connect.redhat.com/kong/kong-ingress-controller:2.5.0-redhat
```
Comment (jrsmroz) [inline config/redhat-certified/related_images.yaml:10]: If the `RELATED_IMAGE_` env vars are used, the `spec.relatedImages` are updated automatically.
However, it seems the certification pipeline want also the rbac proxy and operator images to be included in `spec.relatedImages`

File: config/redhat-certified/related_images.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: kong-gateway-operator.v0.0.0
```
Comment (jrsmroz) [inline config/redhat-certified/related_images.yaml:4]: I propose we start using v0.0.1. I actually made other PR to bump that.

File: config/redhat-certified/related_images.yaml (hunk 1 above)
Comment (pmalek) [reply config/redhat-certified/related_images.yaml]: You're right. I've dropped the `related_images.yaml` patch. This also nullifies the usage of specific version in said patch 👍

File: config/redhat-certified/related_images.yaml (hunk 2 above)
Comment (pmalek) [reply config/redhat-certified/related_images.yaml]: Explained in the other comment. This is not needed anymore.

File: Makefile [hunk 3]
```diff
@@ -34,14 +34,19 @@ endif
 BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 IMAGE_TAG_BASE ?= ghcr.io/kong/gateway-operator
 BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
-BUNDLE_GEN_FLAGS ?= -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+BUNDLE_GEN_FLAGS ?= --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
```
Comment (jrsmroz) [inline Makefile:37]: A question: does that define a bundle version or it's used for defining the operator version.

**Review by jrsmroz — DISMISSED** [review]

File: Makefile (hunk 3 above)
Comment (pmalek) [reply Makefile:37]: This will set `metadata.name`'s version and `spec.version` in CSV.

Comment (pmalek) [reply Makefile:37]: Does that answer your question or do you have any other concerns about it?

**Review by jrsmroz — APPROVED** [review]

**Pull request merged**